### PR TITLE
File size lines

### PR DESCRIPTION
**Pull request by MarkGibbons**
Add a file resource matcher to check the size of a file in lines.

### CODE REVIEW

File: lib/resources/file.rb [hunk 1]
```diff
@@ -44,7 +44,7 @@ def initialize(path)
     %w{
       type exist? file? block_device? character_device? socket? directory?
       symlink? pipe? mode mode? owner owned_by? group grouped_into?
-      link_path linked_to? mtime size selinux_label immutable?
+      link_path linked_to? mtime size size_lines selinux_label immutable?
```
Comment (chris-rock) [inline lib/resources/file.rb:47]: There is no need to add this here, since the implementation is done in this file anyway

**Review by chris-rock — COMMENTED** [review]
Is there a reason why we need this function?

```
describe file('/proc/cpuinfo') do
  its('size_lines') { should be 0 }
end
```
can also be written as:
```
describe file('/proc/cpuinfo') do
  its('content.lines.size') { should cmp 0 }
end
```

File: lib/resources/file.rb (hunk 1 above)
Comment (MarkGibbons) [reply lib/resources/file.rb:47]: Thank you. I didn't know you could do that with content or other matchers inside its.  Even afterwards searching using google I can't find any similar examples.  

It's a minor edge case but is there anyway to put code like this "content ? content.lines.count : 0"  inside the its().  I'd like to return size 0 in the case where a file does not exist.

**Comment by MarkGibbons** [comment]
Not needed.

**Comment by adamleff** [comment]
@MarkGibbons re: your question about defaulting to `0` instead of `nil`:

You wouldn't be able to use the `its` syntax for that without presetting some variables beforehand, and that will likely make the CLI output not very helpful should it fail.

You could try using the `expect`-style syntax instead:

```ruby
describe file('/path/to/file') do
  expect(subject.content.nil? ? 0 : content.lines).to be > 100
end
```

... or something like that.

**Pull request closed without merging**